__tests__/app.test.js

```diff
@@ -340,4 +340,62 @@ describe('GET', () => {
             });
         })
     })
-})
\ No newline at end of file
+})
+
+describe('POST', () => {
+    describe('/api/articles/:article_id/comments', () => {
+        test('status: 201, responds with the posted comment', () => {
+            const newComment =   {
+                username: "icellusedkars",
+                body: "From God we came and to Him we return",
+            }
+            return request(app)
+            .post('/api/articles/1/comments')
+            .send(newComment)
+            .expect(201)
+            .then(({body : responseComment}) => {
+                expect(responseComment).toEqual(
+                    expect.objectContaining({
+                        comment_id: 19,
+                        body: 'From God we came and to Him we return',
+                        article_id: 1,
+                        author: "icellusedkars",
+                        votes: 0,
+                        created_at: expect.any(String),
+                    })  
+                )
+            })
+        })
+        describe('ERROR handling', () => {
+            test('status 400 when user tries to post a comment to an invalid article_id', () =>{
+                const newComment =   {
+                    username: "icellusedkars",
+                    body: "There are more stars in the heaven then grains of sand on the planet",
+                }
+                return request(app)
+                .post('/api/articles/One/comments')
+                .send(newComment)
+                .expect(400)
+                .then(({ body: {msg} }) => {
+                    expect(msg).toBe('Invalid article id');
+                });           
+            })
+            test('status 404 when user tries to post a comment to an article which does not exist', () =>{
+                const newComment =   {
+                    username: "icellusedkars",
+                    body: "The sun is 330000 heavier than the earth",
+                }
+                return request(app)
+                .post('/api/articles/7777/comments')
+                .send(newComment)
+                .expect(404)
+                .then(({ body: {msg} }) => {
+                    expect(msg).toBe('Article does not exist');
+                });           
+            })
+        })
+
+    })
+})
```

Comment:
I dont think you have covered all the test cases here, think about what might happen if we tried to post a comment for a user that doesnt exist, or if the request body didn't have all the right keys. Whenever we give control to the use for there are much more things that could go wrong